lib/active_merchant/billing/gateways/pin.rb

```diff
@@ -158,6 +159,16 @@ def add_metadata(post, options)
         post[:metadata] = options[:metadata] if options[:metadata]
       end
 
+      def add_3ds(post, options)
+        if options[:three_d_secure]
+          post[:three_d_secure] = {}
+          post[:three_d_secure][:version] = options[:three_d_secure][:version] if options[:three_d_secure][:version]
+          post[:three_d_secure][:eci] = options[:three_d_secure][:eci] if options[:three_d_secure][:eci]
+          post[:three_d_secure][:cavv] = options[:three_d_secure][:cavv] if options[:three_d_secure][:cavv]
+          post[:three_d_secure][:transaction_id] = options[:three_d_secure][:ds_transaction_id] || options[:three_d_secure][:xid]
```

Comment:
Would you mind adding a test with a mock `three_d_secure` that has `ds_transaction_id` in order to test this OR as well?

Reply:
Not a problem. On it.

Reply:
added and squashed PR.